src/components/common/PrivateRoute/PrivateRoute.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Navigate } from 'react-router-dom';
+import { useAuth } from '../../../providers/AuthProvider/AuthProvider';
+
+const PrivateRoute = ({ children, redirectTo }) => {
+  const { user } = useAuth();
+
+  if (!user) {
+    return <Navigate to={redirectTo} replace />;
+  }
+
+  return children;
+};
```

Comment:
You could have done this `user ? children: <Navigate to={} replace />`

anyhow above code is readable